experimental/uefi/loader/src/server.rs

```diff
@@ -28,6 +28,17 @@ use proto::{
     UnaryRequest, UnaryResponse,
 };
 
+fn serialize_request(request: UnaryRequest) -> Vec<u8> {
+    // The payload is the request's body prepended with the 8 byte session_id.
+    // This takes adavantage of the session_id's fixed size to avoid needing
+    // to use a key/value pair binary serialization protocol.
+    request
```

Comment:
nit: I think that initialising the vec with the right capacity and then calling `extend_from_slice` twice might be more efficient.

Reply:
Yup, looks like it'd be faster: https://stackoverflow.com/a/63902932

I guess that makes sense, when calling collect rust does not know the size of the final Vec in advance.